src/main/java/com/ksabbak/javaserver/storage/Store.java

```diff
@@ -0,0 +1,47 @@
+package com.ksabbak.javaserver.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Store implements Persistable {
+    private Map<String, String> store;
+
+    public Store(){
+        store = new HashMap<String, String>();
+    }
+
+    @Override
+    synchronized public String create(String key) throws Exception {
+        return create(key, "");
+    }
+
+    @Override
+    synchronized public String create(String key, String value) throws Exception {
+        if (store.get(key) != null) throw new Exception();
+        store.put(key, value);
+        return key;
+    }
+
+    @Override
+    synchronized public String update(String key, String value) throws Exception{
+        if (store.get(key) == null) throw new Exception();
```

Comment:
See above comment, but call it `KeyNotFoundException`.

Reply:
👍